src/processor/operator/scan_node_id.cpp

```diff
@@ -73,6 +74,8 @@ bool ScanNodeID::getNextTuplesInternal(ExecutionContext* context) {
         setSelVector(context, state, startOffset, endOffset);
     } while (outValueVector->state->selVector->selectedSize == 0);
     metrics->numOutputTuple.increase(outValueVector->state->selVector->selectedSize);
+    context->clientContext->progressBar->finishJobsInPipeline(
```

Comment:
I suggest let's not worry about other source operators but only implement for `ScanNodeID` in our initial version.

I would add an `double getProgress()` interface in `PhysicalOperator` and each source operator implement this interface. That means each source operator need to maintain its own progress (usually in shared state). For `ScanNodeID`, you just need to additionally track how many nodeIDs have been scanned. The total number of nodeIDs is available in the table.

If we do this then you most likely do not need `addJobsToPipeline` and `finishJobsInPipeline`